From 88b605eabbd8146fbedea9f8692299931e9e9cc5 Mon Sep 17 00:00:00 2001
From: Patrick Spendrin <ps_ml@gmx.de>
Date: Tue, 29 Apr 2014 01:52:40 +0200
Subject: [PATCH 26/28] fix finding the dummy extractor

the dummy extractor executable is moved into the local build directory and
the setting of the path is adapted to the required windows path separator

REVIEW:117851
---
 src/file/autotest/fileindexingjob/CMakeLists.txt          | 5 ++++-
 src/file/autotest/fileindexingjob/fileindexingjobtest.cpp | 4 ++++
 2 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/src/file/autotest/fileindexingjob/CMakeLists.txt b/src/file/autotest/fileindexingjob/CMakeLists.txt
index 3fc5b2b..486ed67 100644
--- a/src/file/autotest/fileindexingjob/CMakeLists.txt
+++ b/src/file/autotest/fileindexingjob/CMakeLists.txt
@@ -6,7 +6,10 @@ set_target_properties(baloo_file_extractor_dummy
 target_link_libraries(baloo_file_extractor_dummy
   ${KDE4_KDECORE_LIBS}
 )
-
+if(WIN32)
+    set_target_properties(baloo_file_extractor_dummy
+        PROPERTIES RUNTIME_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR})
+endif()
 add_definitions("-DBALOO_TEMP_PATH=\\\"${CMAKE_CURRENT_BINARY_DIR}\\\"")
 kde4_add_unit_test(fileindexingjobtest NOGUI fileindexingjobtest.cpp
                                              ../../fileindexingjob.cpp)
diff --git a/src/file/autotest/fileindexingjob/fileindexingjobtest.cpp b/src/file/autotest/fileindexingjob/fileindexingjobtest.cpp
index 8a0ca84..4e79235 100644
--- a/src/file/autotest/fileindexingjob/fileindexingjobtest.cpp
+++ b/src/file/autotest/fileindexingjob/fileindexingjobtest.cpp
@@ -30,7 +30,11 @@ void FileIndexingJobTest::init()
 {
     // Set the correct path
     QByteArray path = qgetenv("PATH");
+#ifndef Q_OS_WIN
     path = QByteArray(BALOO_TEMP_PATH) + ":" + path;
+#else
+    path = QByteArray(BALOO_TEMP_PATH) + ";" + path;
+#endif
 
     setenv("PATH", path.constData(), 1);
     unsetenv("BALOO_EXTRACTOR_FAIL_FILE");
-- 
1.9.0

